### ISSUE TITLE
com.github.shyiko.mysql.binlog.event.deserialization.EventDataDeserializationException

### ISSUE THREAD

**Issue by alsmn** [shyiko/mysql-binlog-connector-java#158]
# 1. the version of mysql-binlog-connector-java is 0.11.0
        my jdk version is 1.7
        mysql version is 5.6
# 2. my test code : 

public class BinlogUtil implements Serializable{
    public static void main(String[] args) throws IOException{
        String str = "/Users/test/mysqlbinlog-test-0000001";
        File binlogfile = new File(str);
        BinaryLogFileReader reader = new BinaryLogFileReader(binlogfile);
        try{
            for(Event event;(event = reader.readEvent()) != null;){
                System.out.println(event.getData());
            }
        } finally {
            reader.close();
        }
    }
}

# 3. the exception:
Exception in thread "main" com.github.shyiko.mysql.binlog.event.deserialization.EventDataDeserializationException: Failed to deserialize data of EventHeaderV4{timestamp=1488273472000, eventType=EXT_WRITE_ROWS, serverId=2253307, headerLength=19, dataLength=29, nextPosition=9458, flags=0}
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.deserializeEventData(EventDeserializer.java:212)
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.nextEvent(EventDeserializer.java:180)
	at com.github.shyiko.mysql.binlog.BinaryLogFileReader.readEvent(BinaryLogFileReader.java:82)
	at com.xxg.jdeploy.util.BinlogUtil.main(BinlogUtil.java:20)
Caused by: java.io.EOFException
	at com.github.shyiko.mysql.binlog.io.ByteArrayInputStream.read(ByteArrayInputStream.java:190)
	at com.github.shyiko.mysql.binlog.io.ByteArrayInputStream.readInteger(ByteArrayInputStream.java:46)
	at com.github.shyiko.mysql.binlog.event.deserialization.AbstractRowsEventDataDeserializer.deserializeLong(AbstractRowsEventDataDeserializer.java:212)
	at com.github.shyiko.mysql.binlog.event.deserialization.AbstractRowsEventDataDeserializer.deserializeCell(AbstractRowsEventDataDeserializer.java:150)
	at com.github.shyiko.mysql.binlog.event.deserialization.AbstractRowsEventDataDeserializer.deserializeRow(AbstractRowsEventDataDeserializer.java:132)
	at com.github.shyiko.mysql.binlog.event.deserialization.WriteRowsEventDataDeserializer.deserializeRows(WriteRowsEventDataDeserializer.java:64)
	at com.github.shyiko.mysql.binlog.event.deserialization.WriteRowsEventDataDeserializer.deserialize(WriteRowsEventDataDeserializer.java:56)
	at com.github.shyiko.mysql.binlog.event.deserialization.WriteRowsEventDataDeserializer.deserialize(WriteRowsEventDataDeserializer.java:32)
	at com.github.shyiko.mysql.binlog.event.deserialization.EventDeserializer.deserializeEventData(EventDeserializer.java:206)
	... 3 more
# 4. I have seen some issue about EventDataDeserializationException,but i still don't fix it out.

**Comment by shyiko** [comment]
Hi @alsmn. Any chance you can share mysqlbinlog-test-0000001 (over http://transfer.sh/ would be fine)? Also, are there any error while trying to read the same file using `mysqlbinlog`?

**Comment by alsmn** [comment]
Hello @shyiko  .
 I have tried to use mysqlbinlog to read same file, and everything is ok. 
 And I have uploaded the file , the url is https://transfer.sh/dBwW9/mysqlbinlog-test-0000001.zip
Thank you.

**Comment by shyiko** [comment]
Thank you. 

Alright, so the problem is BinaryLogFileReader does not detect checksum type (while BinaryLogClient does). I have created a ticket - https://github.com/shyiko/mysql-binlog-connector-java/issues/160 which should take care of this (I think it has already been fixed in develop branch, though). Until then checksum type has to be specified manually, like so

```java
EventDeserializer eventDeserializer = new EventDeserializer();
eventDeserializer.setChecksumType(ChecksumType.CRC32);
BinaryLogFileReader fileReader = new BinaryLogFileReader(file, eventDeserializer);
...
```

Cheers!
